### ISSUE TITLE
Passing Mode in injectManifest constructor doesn't work

### ISSUE THREAD

**Issue by lizryzhko** [GoogleChrome/workbox#2696]
Welcome! Please use this template for reporting bugs or requesting features. For questions about using Workbox, the best place to ask is Stack Overflow, tagged with `[workbox]`: https://stackoverflow.com/questions/ask?tags=workbox

**Library Affected**:
*workbox-webpack-plugin*

**Browser & Platform**:
*All*

**Issue Description**:
*Passing `Mode` in `InjectManifest` constructor doesn't override the mode value in the current webpack compilation*

**Code**

in webpack config:
  ```
 new WorkboxPlugin.InjectManifest({
                    swSrc: './src/sw.js',
                    swDest: path.join(__dirname, 'public/service-worker.js'),
                    maximumFileSizeToCacheInBytes: 1000000000,
                    include: [/..*(js|css)$/],
                    mode: 'development'
                }),
```
sw.js:
```
import { precacheAndRoute, cleanupOutdatedCaches } from 'workbox-precaching';
import { skipWaiting, clientsClaim, setCacheNameDetails } from 'workbox-core';

setCacheNameDetails({
    prefix: 'my-sw'
});

skipWaiting();
clientsClaim();

// eslint-disable-next-line no-underscore-dangle
const mainfest = self.__WB_MANIFEST;

precacheAndRoute(mainfest);

cleanupOutdatedCaches();
```
**Result**
Expected: I can see logs in the console
Actual: No logs

** Versions **
"workbox-webpack-plugin": "6.0.2",
"webpack": "4.44.2"

**Comment by jeffposnick** [comment]
Thanks for letting us know. I had thought that it worked, but upon closer inspection, you're correct, and we didn't have that scenario covered by our test suite 😦 

I just played around with a few changes to the codebase, and I'm actually not sure if it's possible to override the parent compilation mode from within a child compilation. Nothing I tried (including just manually applying a `DefinePlugin` instance with a custom `process.env.NODE_ENV` value to the child compilation) worked.

So, this might end up being a case where the parameter is marked as being unsupported, and where the `mode` of the parent compilation needs to be toggled to control the `InjectManifest` behavior.

Sorry for any inconvenience or wasted time this might have incurred!
